src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java

```diff
@@ -41,25 +41,27 @@ public SensitivityAnalysisInputBuilderService(ActionsService actionsService, Fil
         this.filterService = filterService;
     }
 
-    private List<Contingency> goGetContingencies(EquipmentsContainer contingencyListIdent, UUID networkUuid, String variantId, ReportNode reporter) {
-        try {
-            return actionsService.getContingencyList(contingencyListIdent.getContainerId(), networkUuid, variantId);
-        } catch (Exception ex) {
-            LOGGER.error("Could not get contingencies from " + contingencyListIdent.getContainerName(), ex);
+    private List<Contingency> goGetContingencies(List<EquipmentsContainer> contingencyListIdent, UUID networkUuid, String variantId, ReportNode reporter) {
+        List<UUID> ids = contingencyListIdent.stream().map(EquipmentsContainer::getContainerId).toList();
+        Contingencies contingencies = actionsService.getContingencyList(ids, networkUuid, variantId);
+        if (contingencies == null) {
+            return List.of();
+        }
+
+        contingencies.getContingenciesNotFound().forEach(id -> {
+            EquipmentsContainer container = contingencyListIdent.stream().filter(c -> c.getContainerId().equals(id)).findFirst().orElseThrow();
```

Comment:
Could we find instead of filter then findFirst here ?

Reply:
i dont know what you mean , because it's a list we have to iterate over all elements i think